### ISSUE TITLE
Allow default port in clients

### ISSUE THREAD

**Issue by skunert** [paritytech/jsonrpsee#1048]
Currently users have to pass a `impl AsRef<str>` to the client builder that includes a port. 

I know there have been other issues (https://github.com/paritytech/jsonrpsee/issues/554, https://github.com/paritytech/subxt/issues/644) about this, and it is marked as blocked on https://github.com/hyperium/http/issues/509. But I would still like to advocate once more for jsonrpsee handling of a default port when passing a URL to it.

As a library user, I expect that I could pass a valid URL to jsonrpsee. The [URL standard](https://url.spec.whatwg.org/#port-state) says that the port should be omitted if it is the default port. This is also what users expect.

The problem is further amplified by rust `url` and `http` crates. If an application takes user input and parses it to an `url::Url`, the default port gets lost, and there is no nice way to force the crate to add it back when converting to string. `http::Uri` does not remove the default port, but is not able to provide the default port for different schemes (`Url` has at least [port_or_known_default](https://docs.rs/url/latest/url/struct.Url.html#method.port_or_known_default)) if none was provided.

**Ideas:**
- Maybe jsonrpsee could do manual port handling that does not rely on https://github.com/hyperium/http/issues/509 (e.g. extracting it with a regex).
- Alternatively we could maybe have a builder method `build_with_components(scheme, host, port, path ...)` which would add some convenience

**Comment by niklasad1** [comment]
Fair enough, I don't expect that issue to resolved any time soon so better provide some solution around it
